### PR TITLE
Remove Map null annotation workarounds

### PR DESCRIPTION
**Pull request by wborn**
These workarounds to prevent false positives can be removed now the EEAs allow for proper null analysis.

Depends on https://github.com/openhab/openhab-core/pull/1780

### CODE REVIEW

**Comment by wborn** [comment]
Since https://github.com/openhab/openhab-core/pull/1780 got merged, it will be soon available. So it would be nice if this PR can also be reviewed, so it can merged without builds being broken too long.

**Comment by robnielsen** [comment]
Insteon changes lgtm

**Review by robnielsen — APPROVED** [review]

File: bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/AccountServlet.java
Comment (J-N-K) [inline bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/AccountServlet.java:559]: Did you check that the array can be null but all array contents are non-null? I would expect that the array is nonnull but it contents are nullable, so `@Nullable Playlist[]` would be correct.

File: bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/AmazonEchoDynamicStateDescriptionProvider.java
Comment (J-N-K) [inline bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/AmazonEchoDynamicStateDescriptionProvider.java:142]: see above

File: bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/jsons/JsonMusicProvider.java
Comment (J-N-K) [inline bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/jsons/JsonMusicProvider.java:28]: From my understanding this is not a workaround but reflects the null-values from the JSON response (also immediately below)

File: bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/jsons/JsonPlaylists.java
Comment (J-N-K) [inline bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/jsons/JsonPlaylists.java:28]: I'm also not sure here, it could be that the json response contains null-values in null-arrays

**Review by J-N-K — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/AccountServlet.java
Comment (wborn) [reply bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/AccountServlet.java:559]: They are parsed from JSON with Gson so I would expect the array map value to be either non null or nullable with non null array elements. I would think it's using the correct [syntax](https://help.eclipse.org/2020-09/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Ftasks%2Ftask-using_null_type_annotations.htm&cp%3D1_3_9_1_3_0&anchor=compatibility_syntax)?

```java
    // annotated array type:
    Object @NonNull[] o2;
    o2 = null;          // NOT OK
    o2 = new Object[1];
    o2[0] = null;       // OK
```

Comment (J-N-K) [reply bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/AccountServlet.java:559]: Mh. Good point. Then the annotation was wrong before (and not work-around related (because the members of the array are not what determines the result of `Map.get` but the array itself.

**Review by J-N-K — APPROVED** [review]

**Comment by J-N-K** [comment]
@openhab/add-ons-maintainers  I would like to merge this even if we don't get jenkins to succeed. Since the core PR has been merged, I can't work on #8949 because I would have to doible the work from here.

**Comment by wborn** [comment]
Yes let's merge it because the build will be broken without these changes.
I promise to fix anything that breaks because of it. :-)

**Comment by kaikreuzer** [comment]
With such a promise, let's merge all PRs ;-)

**Pull request merged**
